pkg/config/config_test.go

```diff
@@ -1054,3 +1060,121 @@ func TestWaitForPodsReadyIsEnabled(t *testing.T) {
 		})
 	}
 }
+
+func controlOptionsCacheWithClusterProfile(namespace string) ctrlcache.Options {
+	cOpts := defaultControlOptions(namespace)
+	cOpts.Cache.ByObject[objectKeyClusterProfile] = ctrlcache.ByObject{
+		Namespaces: map[string]ctrlcache.Config{
+			namespace: {},
+		},
+	}
+	return cOpts.Cache
+}
+
+func TestConfigureClusterProfileCacheWithClient(t *testing.T) {
+	ctx := context.Background()
+	log := logr.Discard()
```

Comment:
Or probably my better:

```suggestion
	ctx, log := utiltesting.ContextWithLog(t)
```